src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1121,7 +1114,7 @@ LoadVectorNode* LoadVectorNode::make(int opc, Node* ctl, Node* mem,
 Node* LoadVectorNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   const TypeVect* vt = vect_type();
   if (Matcher::vector_needs_partial_operations(this, vt)) {
-    return VectorNode::try_to_gen_masked_vector(phase, this, vt);
+    return VectorNode::gen_masked_vector(phase, this, vt);
   }
```

Comment:
I think it would still be good practice to expect that a `nullptr` could come from `gen_masked_vector`, and then continue with optimizations below, rather than just returning the `nullptr`.

Because: who knows what someone in the future might do inside `gen_masked_vector`, maybe they'll find some edge case and just return `nullptr` again, and then we are back to similar issues.

Reply:
I see your concern. Make sense to me. Thanks! I'd like keep current implementation of `Matcher::vector_needs_partial_operations` and `gen_masked_vector` because as we discussed in the previous PR that this sounds more reasonable. 

I will update the caller code here to check `nullptr` in addition although it won't generate a `nullptr` now. Code may look like:
```
if (Matcher::vector_needs_partial_operations(this, vt)) {
    Node* n = VectorNode::gen_masked_vector(phase, this, vt);
    if (n != nullptr) {
      return n;
    }
 }
return ...
```

Reply:
Is it better that we add an assertion of non `nullptr` value before returning in `gen_masked_vector` , consider this might make the caller code clean?

Reply:
Sounds good to me. I will change the code by combining the methods into a function. Thanks!